Code/Framework/AzCore/AzCore/PlatformDef.h

```diff
@@ -16,13 +16,16 @@
 
 #if defined(__clang__)
     #define AZ_COMPILER_CLANG   __clang_major__
+    #define AZ_TRAIT_COMPILER_TEMPLATE_EXTERN extern
 #elif defined(__GNUC__)
     //  Assign AZ_COMPILER_GCC to a number that represents the major+minor (2 digits) + path level (2 digits)  i.e. 3.2.0 == 30200
     #define AZ_COMPILER_GCC     (__GNUC__ * 10000 \
                                + __GNUC_MINOR__ * 100 \
                                + __GNUC_PATCHLEVEL__)
+    #define AZ_TRAIT_COMPILER_TEMPLATE_EXTERN extern
 #elif defined(_MSC_VER)
     #define AZ_COMPILER_MSVC    _MSC_VER
+    #define AZ_TRAIT_COMPILER_TEMPLATE_EXTERN
```

Comment:
So on MSVC compilers, you don't extern templates using extern?

Reply:
Yes, this is why
https://learn.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-1-c4910?view=msvc-170

Reply:
Ah, I see.  Might be helpful to indicate that in the file there, for future maintainers.  From my reading of this, you still need to use extern on templates in MSVC if you are NOT dll exporting them.  So for example, monolithic mode?  But then again in monolithic mode, you don't need to extern them at all, becuase everythings linked into one file.